internal/settings.go

```diff
@@ -19,14 +19,16 @@ import (
 )
 
 const (
-	newAuthLibEnVar = "GOOGLE_API_GO_EXPERIMENTAL_USE_NEW_AUTH_LIB"
+	newAuthLibEnVar       = "GOOGLE_API_GO_EXPERIMENTAL_USE_NEW_AUTH_LIB"
+	universeDomainDefault = "googleapis.com"
 )
 
 // DialSettings holds information needed to establish a connection with a
 // Google API service.
 type DialSettings struct {
 	Endpoint                      string
 	DefaultEndpoint               string
+	DefaultEndpointTemplate       string
```

Comment:
Until such time as service configs provide the template format, I think this should be derived rather than a property of DialSettings.  Or do you envision the generators computing and supplying this at codegen time?

Reply:
Yes this is an internaloption to be set by the generator

Reply:
Also, I found using `mergeEndpoints` to be a bit buggy when using it to replace the universe domain.